Hunting Queries/MultipleDataSources/RareDNSLookupWithDataTransfer.txt

```diff
@@ -0,0 +1,96 @@
+CommonLookups
+)
+on Domain;
+// Join back the Daily Average Lookup Count to add context to rarity over lookback period
+let RareLookups = UncommonLookupsToday | join kind= innerunique (
+DomainLookups 
+) on Domain 
+| project LookupStartTime, LookupEndTime, ClientIP, Domain, IPAddresses, LookupCountToday, DailyAvgLookupCountOverLookback;
+let DNSIPBreakout = RareLookups
+| extend DnsIPAddress = iff(IPAddresses has ",", split(IPAddresses, ","), todynamic(IPAddresses)) 
+| mvexpand DnsIPAddress
+| extend DnsIPAddress = tostring(DnsIPAddress)
+| distinct LookupStartTime, LookupEndTime, ClientIP, Domain, DnsIPAddress, LookupCountToday, DailyAvgLookupCountOverLookback
+| extend DnsIPType = iff(DnsIPAddress matches regex PrivateIPregex,"private" ,"public" )
+| where DnsIPType =="public"
+;
+let DataMovement = ( union isfuzzy=true 
+(CommonSecurityLog
+| where TimeGenerated >= endtime
+| where DeviceVendor =="Palo Alto Networks" and Activity == "TRAFFIC"
+| extend DestinationIPType = iff(DestinationIP matches regex PrivateIPregex,"private" ,"public" )
+| where DestinationIPType =="public"
+| project DataType = DeviceVendor, TimeGenerated, SourceIP , SourcePort , DestinationIP, DestinationPort, ReceivedBytes, SentBytes
+| sort by SourceIP asc, SourcePort asc,TimeGenerated asc, DestinationIP asc, DestinationPort asc
+| summarize sum(ReceivedBytes), sum(SentBytes), count() by DataType, SourceIP , SourcePort , DestinationIP, DestinationPort
+| sort by sum_SentBytes desc
+),
+(WireData
+| where TimeGenerated >= endtime
+| where Direction == "Outbound"
+| extend RemoteIPType = iff(RemoteIP matches regex PrivateIPregex,"private" ,"public" ) 
+| where RemoteIPType =="public" 
+| project DataType = Type, TimeGenerated , SourceIP = LocalIP , SourcePort = LocalPortNumber , DestinationIP = RemoteIP, DestinationPort = RemotePortNumber, ReceivedBytes, SentBytes 
+| summarize sum(ReceivedBytes), sum(SentBytes), count() by DataType, SourceIP , SourcePort , DestinationIP, DestinationPort
+| extend DataType = Type
+| sort by sum_SentBytes desc
+),
+(VMConnection 
+| where TimeGenerated >= endtime
+| where Direction == "outbound"
+| extend DestinationIPType = iff(DestinationIp matches regex PrivateIPregex,"private" ,"public" )
+| where DestinationIPType =="public"
+| project DataType = Type, TimeGenerated , SourceIP = SourceIp , DestinationIP = DestinationIp, DestinationPort, ReceivedBytes = BytesReceived, SentBytes = BytesSent 
+| summarize sum(ReceivedBytes), sum(SentBytes), count() by DataType, SourceIP , DestinationIP, DestinationPort
+| sort by sum_SentBytes desc
+)
+);
+DNSIPBreakout | join kind = leftouter (
+DataMovement
+) on $left.DnsIPAddress == $right.DestinationIP and $left.ClientIP == $right.SourceIP
+// The below condition can be removed to see all DNS results.
+// This is used here as the goal of the query is to connect rare DNS lookups to a data type that can show byte transfers to that given DestinationIP
+| where isnotempty(DataType)
```

Comment:
project-away ClientIP (by definition of the join this is always the same as SourceIP)
Ditto project-away DNS IP? (covered by DestinationIP ?)

Reply:
Yep - added | project-away DnsIPAddress, ClientIP